src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -1,21 +1,127 @@
 #include "ResourceGatheringOperation.hpp"
+#include <utility>
+#include "openvic-simulation/pop/Pop.hpp"
+#include "openvic-simulation/map/ProvinceInstance.hpp"
+#include "openvic-simulation/map/TerrainType.hpp"
+#include "openvic-simulation/modifier/Modifier.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/economy/production/ProductionType.hpp"
 
 using namespace OpenVic;
 
 ResourceGatheringOperation::ResourceGatheringOperation(
-	ProductionType const& new_production_type,
+	ProductionType const* new_production_type,
 	fixed_point_t new_size_multiplier,
 	fixed_point_t new_revenue_yesterday,
 	fixed_point_t new_output_quantity_yesterday,
 	fixed_point_t new_unsold_quantity_yesterday,
-	ordered_map<Pop*, Pop::pop_size_t>&& new_employees
+	ordered_map<Pop const*, Pop::pop_size_t>&& new_employees
```

Comment:
I had to change this to const as province instance doesn't expose mutable pops.
We'll have to change province instance to expose them as they must be mutable, so the RGO can pay them.

Reply:
Turned out I have to use a different iterator:
`for (plf::colony<Pop>::iterator it = pops.begin(); it != pops.end(); ++it)`